mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/services/TiltaksgjennomforingService.kt

```diff
@@ -110,6 +122,7 @@ class TiltaksgjennomforingService(
 
         return tiltaksgjennomforingRepository
             .delete(id)
+            .onRight { arenaMigreringTiltaksgjennomforingKafkaProducer.retract(id) }
```

Comment:
Dette er nok et problem vi har skjøvet litt på, men kan hende er noe vi burde se på ifm. denne featuren.

Hvis `.delete(id)` kjører OK, men denne linjen feiler så kan vi ende opp i tilstand der vi deler forskjellig data på kafka fra det vi faktisk har i databasen. Dette gjelder også for `upsert`-funksjonen over (vi kan ende opp med en tilstand der upsert til databasen blir gjort, men et av stegene etter feiler/eller flere ikke blir utført pga. tidligere nevnte feil).

Kafka er jo asynkront, så innhold på topic vil aldri matche database hele tiden, men vi burde ha noen mekanismer på plass som sørger for at dette henter seg inn igjen (i tillegg så burde vi jo også sørge for at andre ting som utkast og data i sanity også henger sammen med dataen).

Et alternativ kan være å kjøre alt i én databasetransaksjon, noe ala:
1. Start databasetransaksjon
2. Slett gjennomføring fra db og kafka
3. Commit tranasksjon, eller rull tilbake om publish til kafka feilet
4. Og noe tilsvarende transaksjonslogikk for upsert-funksjonen.

Et annet alternativ kan kanskje være å kjøre noen ekstra rutine-jobber som sørger for å holde ting i sync, men det kan potensielt være litt slitsomt å håndtere alle disse mulige tilstandene "after the fact" heller enn "up front" (har ikke helt tenkt gjennom forskjellige alternativer for dette enda). 

(Vi har jo også noe lignene upsert/delete-funksjonalitet i `ArenaAdapterService`, men her skal oppførsel være ivaretatt siden upsert/delete er idempotent _og_ `arena-adapter` har en retry-mekanimse som trigger endepunktet på nytt så lenge _noe_ har feilet.)

Reply:
Godt poeng. Synes å kjøre i en transaksjon høres best ut umiddelbart.